extensions/arc/src/ui/dashboards/postgres/postgresParameters.ts

```diff
@@ -147,21 +147,20 @@ export abstract class PostgresParametersPage extends DashboardPage {
 							cancellable: false
 						},
 						async (_progress, _token): Promise<void> => {
+							let session: azdataExt.AzdataSession | undefined = undefined;
 							try {
 								this.parameterUpdates!.forEach((value, key) => {
 									engineSettings.push(`${key}="${value}"`);
 								});
-								const session = await this._postgresModel.controllerModel.acquireAzdataSession();
-								try {
-									this.saveParameterEdits(engineSettings.toString(), session);
-								} finally {
-									session.dispose();
-								}
+								session = await this._postgresModel.controllerModel.acquireAzdataSession();
+								await this.saveParameterEdits(engineSettings.toString(), session);
```

Comment:
Good find, I expect the cause of the uncaught exceptions is that these calls weren't awaited.

Reply:
Yeah, that'd be the issue. WIthout the awaits the thread continues on and so the catch does nothing